### ISSUE TITLE
Error: [commonjs] Failed to resolve entry for package "@jimp/custom". The package may have incorrect main/module/exports specified in its package.json.

### ISSUE THREAD

**Issue by usernein** [nut-tree/nut.js#513]
Hello, i'm getting this error while trying to use nut-js with TypeScript and Vite:
```log
[commonjs] Failed to resolve entry for package "@jimp/custom". The package may have incorrect main/module/exports specified in its package.json.
file: /home/dev/workspace/mm/pje-robot/node_modules/@nut-tree/nut-js/dist/lib/mouse.class.js
```

**Desired execution environment / tested on**

- [ ] Virtual machine
- [ ] Docker container
- [x] Dev/Host system

The code i'm trying to run is very simple:
```js
import { store } from './store'
import { keyboard, Key } from '@nut-tree/nut-js'

export enum MacroInsertInto {
  OWING,
  PAID
}

export const insertIntoCells = (valueOption: MacroInsertInto) => {
  const values =
    valueOption == MacroInsertInto.OWING ? store.owingValues : store.paidValues

  for (const value of values) {
    const stripped = value.replace(/\D/g, '')
    for (let i = 0; i < stripped.length; i++) {
      keyboard.type(stripped[i])
    }

    keyboard.type(Key.Down)
  }
}
```

**Comment by usernein** [comment]
Maybe it has something to do with https://github.com/jimp-dev/jimp/issues/1110#issue-1458465740?

**Comment by wimurk** [comment]
I have the same issue. @usernein did you solve the issue?

**Comment by wimurk** [comment]
Current version is still using the old 0.16 verison of jimp. @s1hofmann when can we expect eh new upated version

**Comment by s1hofmann** [comment]
@wimurk This change contains at least one breaking change so it’ll be shipped with a major release. A release date for this has not been set.

**Comment by wimurk** [comment]
> @wimurk This change contains at least one breaking change so it’ll be shipped with a major release. A release date for this has not been set.

Any workaround on this? The package isn't usable with this issue.

**Comment by wimurk** [comment]
> > @wimurk This change contains at least one breaking change so it’ll be shipped with a major release. A release date for this has not been set.
> 
> Any workaround on this? The package isn't usable with this issue.

For anyone having the same problem. You can fix this issue in yarn with resolutions.

```json
"resolutions" : {
      "jimp": "^0.22.8"
  }
```

This will force every package to use this version.

**Comment by s1hofmann** [comment]
> > @wimurk This change contains at least one breaking change so it’ll be shipped with a major release. A release date for this has not been set.
> 
> Any workaround on this? The package isn't usable with this issue.

I beg to differ. It’s currently not usable out of the box  with setups that are using Vite. Since this has only been the second report for this issue it’s not my highest priority.

Thanks for posting your workaround!

All the best 

Simon

**Comment by i-am-jabriel** [comment]
Is there still no workaround for this?

**Comment by s1hofmann** [comment]
@i-am-jabriel [A rworkaround has been posted ](https://github.com/nut-tree/nut.js/issues/513#issuecomment-1621794839)

**Comment by s1hofmann** [comment]
Closed with [release 4.0.0](https://github.com/nut-tree/nut.js/releases/tag/v4.0.0)
